dev/com.ibm.ws.cdi.third.party_fat/test-applications/CDIInEntityListernersTestEarLib.jar/src/cdi/entity/listeners/test/model/lib/EntityBListener.java

```diff
@@ -0,0 +1,45 @@
+/*******************************************************************************
+ * Copyright (c) 2019 IBM Corporation and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *     IBM Corporation - initial API and implementation
+ *******************************************************************************/
+
+package cdi.entity.listeners.test.model.lib;
+
+import javax.enterprise.inject.spi.BeanManager;
+import javax.inject.Inject;
+
+import javax.persistence.Entity;
+import javax.persistence.GeneratedValue;
+import javax.persistence.GenerationType;
+import javax.persistence.Id;
+import javax.persistence.PostPersist;
+import javax.persistence.PrePersist;
+
+public class EntityBListener {
+
+    @Inject 
+    BeanManager bm;
+   
+    @PrePersist
+    public void prePersist(Object object) {
+        if (bm != null) {
+            System.out.println("prePersist" + object.toString() + " " + bm.toString());
+            System.out.println("testInjectWorksInsideEntityListeners passed!");
+        }
+    }
+   
+    @PostPersist
+    public void postPersist(Object object){
+        if (bm != null) {
+            System.out.println("postPersist" + object.toString() + " " + bm.toString());
+            System.out.println("testInjectWorksInsideEntityListeners passed!");
```

Comment:
Shouldn't these two messages be different?

Reply:
i.e. the one in prePersist should be different to the one in postPersist

Reply:
Not particularly. All I'm really doing is checking that injection works properly into an EntityListener that lives in an ear lib. So it doesn't really matter which method I used, putting it in both was more a "why not" than a "why".

Reply:
ok